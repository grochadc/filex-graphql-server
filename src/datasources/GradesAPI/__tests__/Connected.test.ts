import { GradesAPI } from "../index";

import * as dotenv from "dotenv";
dotenv.config({
  path: ".env.test",
});

import { PrismaClient } from "@prisma/client";

const prisma = new PrismaClient({
  datasources: {
    db: {
      url: process.env.DATABASE_URL,
    },
  },
});

if (process.env.DOTENV_ENV !== "test") {
  console.log("Init prisma with url", process.env.HEROKU_POSTGRESQL_TEAL_URL);
  throw new Error(
    `Careful! If this script runs with a production db string reservations will be erased at the end!!/nCheck the file .env.test is being read by Jest instead of .env`
  );
}

const gradesAPI = new GradesAPI(prisma);

test("works", async () => {
  const data = await gradesAPI.postGrades(
    [
      { codigo: "1234567890", finalGrade: "100", situacion: "A" },
      { codigo: "0987654321", finalGrade: "90", situacion: "A" },
    ],
    "2022B",
    "en"
  );
  expect(data).toBeDefined();
  expect(data).toMatchSnapshot();
  console.log("data", data);
});

test("crashes as expected", async () => {
  await expect(
    gradesAPI.postGrades(
      [{ codigo: "1234509876", finalGrade: "60", situacion: "NA" }],
      "2022B",
      "en"
    )
  ).rejects.toThrow('Record to update not found.');
});
