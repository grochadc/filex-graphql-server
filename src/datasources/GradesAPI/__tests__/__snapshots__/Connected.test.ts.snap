// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
[
  {
    "applicantId": 1,
    "ciclo_actual": "2022B",
    "codigo": "1234567890",
    "curso": "en",
    "desertor": false,
    "finalGrade": "100",
    "groupId": 2,
    "id": 3,
    "nivel": 4,
    "situacion": "A",
  },
  {
    "applicantId": 2,
    "ciclo_actual": "2022B",
    "codigo": "0987654321",
    "curso": "en",
    "desertor": false,
    "finalGrade": "90",
    "groupId": 2,
    "id": 2,
    "nivel": 4,
    "situacion": "A",
  },
]
`;
